crates/symbolicator/src/services/symbolication.rs

```diff
@@ -1902,11 +2118,82 @@ fn stackwalk_with_breakpad(
+    cfi_caches: BTreeMap<DebugId, PathBuf>,
+    minidump_path: PathBuf,
+    spawn_time: SystemTime,
 ) -> Result<procspawn::serde::Json<StackWalkMinidumpResult>, ProcError> {
-    unimplemented!()
+    if let Ok(duration) = spawn_time.elapsed() {
+        metric!(timer("minidump.stackwalk.spawn.duration") = duration);
+    }
+
+    // Stackwalk the minidump.
+    let minidump = Minidump::read(ByteView::open(minidump_path)?)?;
+    let provider = TempSymbolProvider::new(cfi_caches.iter());
+    let duration = Instant::now();
+    let process_state = minidump_processor::process_minidump(&minidump, &provider)?;
+    let duration = duration.elapsed();
+
+    let minidump_state = MinidumpState::from_rust_minidump(&process_state);
+    let object_type = minidump_state.object_type();
+
+    let modules: BTreeMap<DebugId, RawObjectInfo> = process_state
+        .modules
+        .iter()
+        .map(|module| {
+            (
+                // TODO(ja): Check how this can be empty and how we shim.
+                //           Probably needs explicit conversion from raw
+                DebugId::from_str(&module.debug_identifier().unwrap_or_default())
+                    .unwrap_or_default(),
+                object_info_from_minidump_module_rust_minidump(object_type, module),
+            )
+        })
+        .collect();
+
+    // Finally iterate through the threads and build the stacktraces to
+    // return, marking modules as used when they are referenced by a frame.
+    let requesting_thread_index: Option<usize> = process_state.requesting_thread;
+    let threads = process_state.threads;
+    let mut stacktraces = Vec::with_capacity(threads.len());
+    for (index, thread) in threads.iter().enumerate() {
+        let registers = match thread.frames.get(0) {
+            Some(frame) => map_symbolic_registers_rust_minidump(&frame.context),
+            None => Registers::new(),
+        };
+
+        // Trim infinite recursions explicitly because those do not
+        // correlate to minidump size. Every other kind of bloated
+        // input data we know is already trimmed/rejected by raw
+        // byte size alone.
+        let frame_count = thread.frames.len().min(20000);
+        let mut frames = Vec::with_capacity(frame_count);
+        for frame in thread.frames.iter().take(frame_count) {
+            frames.push(RawFrame {
+                instruction_addr: HexValue(frame.return_address),
+                package: frame.module.as_ref().map(|m| m.code_file().into_owned()),
+                trust: frame.trust,
+                ..RawFrame::default()
+            });
+        }
+
+        stacktraces.push(RawStacktrace {
+            is_requesting: requesting_thread_index.map(|r| r == index),
+            thread_id: Some(thread.thread_id.into()),
+            registers,
+            frames,
+        });
+    }
+
+    let mut missing_modules = modules.clone();
+    missing_modules.retain(|id, _| provider.missing_ids().contains(id));
```

Comment:
you need to bind `missing_ids()` to a variable. Otherwise this will take (and drop) the Set on the first call, and all the other calls will *always* return an empty Set.

I also wonder if we can combine these in a smarter way, instead of clone+retain.
I have started refactoring this in #662 so we can also do that as a followup.